packages/fast-check/src/arbitrary/_internals/implementations/SchedulerImplem.ts

```diff
@@ -157,37 +157,34 @@ export class SchedulerImplem<TMetaData> implements Scheduler<TMetaData> {
     let resolveSequenceTask = () => {};
     const sequenceTask = new Promise<void>((resolve) => (resolveSequenceTask = resolve));
 
-    sequenceBuilders
-      .reduce((previouslyScheduled: PromiseLike<any>, item: SchedulerSequenceItem<TMetaData>) => {
-        const [builder, label, metadata] =
-          typeof item === 'function' ? [item, item.name, undefined] : [item.builder, item.label, item.metadata];
-        return previouslyScheduled.then(() => {
-          // We schedule a successful promise that will trigger builder directly when triggered
-          const scheduled = this.scheduleInternal(
-            'sequence',
-            label,
-            dummyResolvedPromise,
-            metadata,
-            customAct || defaultSchedulerAct,
-            () => builder(),
-          );
-          scheduled.catch(() => {
-            status.faulty = true;
-            resolveSequenceTask();
-          });
-          return scheduled;
-        });
-      }, dummyResolvedPromise)
-      .then(
-        () => {
-          status.done = true;
-          resolveSequenceTask();
-        },
-        () => {
-          /* Discarding UnhandledPromiseRejectionWarning */
-          /* No need to call resolveSequenceTask as it should already have been triggered */
-        },
-      );
+    const onDone = () => {
+      status.done = true;
+      resolveSequenceTask();
+    };
+    const onFaulty = () => {
+      status.faulty = true;
+      resolveSequenceTask();
+    };
+
+    let previouslyScheduled = dummyResolvedPromise;
```

Comment:
Used to be the initial version received by the reduce function. In other words, the first value of `previouslyScheduled` was `dummyResolvedPromise` (see https://github.com/dubzzz/fast-check/pull/4714/files#diff-dfe3a426c4b614c4cc2b0327598e960f3175233178422e0a3e3acaf331df50f6L180).